src/instructlab/sdg/utils/docprocessor.py

```diff
@@ -0,0 +1,595 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+from typing import List
+import json
+import logging
+import re
+
+from pathlib import Path
+from typing import List, Tuple
+import json
+import re
+
+# Third Party
+from docling.datamodel.base_models import PipelineOptions
+from docling.document_converter import DocumentConverter
+from langchain_text_splitters import Language, RecursiveCharacterTextSplitter
+
+# Third Party
+from datasets import Dataset, concatenate_datasets
+from tabulate import tabulate
+from transformers import AutoTokenizer
+import yaml
+
+# Local
+
+logger = logging.getLogger(__name__)
+DOC_FILEPATH = Path("~/.local/share/instructlab/documents").expanduser()
```

Comment:
This path should probably be somewhere under the `output_dir` from generate_data.py. Since we'll create new docs here for every run, perhaps under the `node_datasets_*` subdirectory for that run?